File: sample/me/xiaopan/easy/imageloader/sample/MyApplication.java
/*
 * Copyright 2013 Peng fei Pan
 * 
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 *   http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package me.xiaopan.easy.imageloader.sample;

import me.xiaoapn.easy.imageloader.ImageLoader;
import me.xiaoapn.easy.imageloader.R;
import me.xiaoapn.easy.imageloader.display.ZoomInBitmapDisplayer;
import me.xiaoapn.easy.imageloader.display.ZoomOutBitmapDisplayer;
import me.xiaoapn.easy.imageloader.process.ReflectionBitmapProcessor;
import me.xiaoapn.easy.imageloader.task.Options;
import android.app.Application;

public class MyApplication extends Application {

	@Override
	public void onCreate() {
		super.onCreate();
		
		ImageLoader.getInstance().init(getBaseContext());
		ImageLoader.getInstance().getConfiguration().setDebugMode(true);
		
		Options defaultOptions = ImageLoader.getInstance().getConfiguration().getDefaultOptions();
		defaultOptions.setLoadingDrawable(getResources(), R.drawable.image_loading);	//设置加载中显示的图片
		defaultOptions.setFailureDrawable(getResources(), R.drawable.image_load_failure); 	//设置加载失败时显示的图片
		defaultOptions.setBitmapDisplayer(new ZoomOutBitmapDisplayer());
		defaultOptions.setBitmapProcessor(new ReflectionBitmapProcessor());
		
		Options viewPagerOptions = defaultOptions.copy();
		viewPagerOptions.setBitmapProcessor(new ReflectionBitmapProcessor());
		viewPagerOptions.setLoadingDrawable(null);
		ImageLoader.getInstance().getConfiguration().putOptions(OptionsType.VIEW_PAGER, viewPagerOptions);
		
		Options listViewOptions = defaultOptions.copy();
		listViewOptions.setBitmapProcessor(new ReflectionBitmapProcessor());
		listViewOptions.setBitmapDisplayer(new ZoomInBitmapDisplayer());
		ImageLoader.getInstance().getConfiguration().putOptions(OptionsType.LIST_VIEW, listViewOptions);
		
		Options galleryOptions = defaultOptions.copy();
		galleryOptions.setBitmapProcessor(new ReflectionBitmapProcessor());
		galleryOptions.setBitmapDisplayer(new ZoomOutBitmapDisplayer());
		ImageLoader.getInstance().getConfiguration().putOptions(OptionsType.GALLERY, galleryOptions);
		
		Options simpleOptions = defaultOptions.copy();
		simpleOptions.setBitmapProcessor(new ReflectionBitmapProcessor());
		simpleOptions.setBitmapDisplayer(new ZoomOutBitmapDisplayer());
		simpleOptions.getCacheConfig().setCacheInMemory(false);
		ImageLoader.getInstance().getConfiguration().putOptions(OptionsType.SIMPLE, simpleOptions);
	}
}

File: sample/src/androidMain/kotlin/com/github/panpf/sketch/sample/MyApplication.kt
/*
 * Copyright (C) 2024 panpf <panpfpanpf@outlook.com>
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.github.panpf.sketch.sample

import android.annotation.SuppressLint
import android.app.Application
import android.content.Context
import com.github.panpf.sketch.SingletonSketch
import com.github.panpf.sketch.Sketch
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.cancel
import java.security.SecureRandom
import java.security.cert.X509Certificate
import javax.net.ssl.HttpsURLConnection
import javax.net.ssl.SSLContext
import javax.net.ssl.TrustManager
import javax.net.ssl.X509TrustManager

class MyApplication : Application(), SingletonSketch.Factory {

    private val coroutineScope = CoroutineScope(Dispatchers.Main)

    override fun onCreate() {
        super.onCreate()
        handleSSLHandshake()
    }

    override fun createSketch(context: Context): Sketch {
        return newSketch(context)
    }

    override fun onTerminate() {
        super.onTerminate()
        coroutineScope.cancel()
    }

    /**
     * for api.pexels.com on Android 5.0
     */
    private fun handleSSLHandshake() {
        try {
            val trustAllCerts = arrayOf<TrustManager>(
                @SuppressLint("CustomX509TrustManager")
                object : X509TrustManager {
                    override fun getAcceptedIssuers(): Array<X509Certificate?> {
                        return arrayOfNulls(0)
                    }

                    @SuppressLint("TrustAllX509TrustManager")
                    override fun checkClientTrusted(
                        certs: Array<X509Certificate?>?,
                        authType: String?
                    ) {
                    }

                    @SuppressLint("TrustAllX509TrustManager")
                    override fun checkServerTrusted(
                        certs: Array<X509Certificate?>?,
                        authType: String?
                    ) {
                    }
                })
            val sc = SSLContext.getInstance("TLS")
            // trustAllCerts trust all certificates
            sc.init(null, trustAllCerts, SecureRandom())
            HttpsURLConnection.setDefaultSSLSocketFactory(sc.socketFactory)
            HttpsURLConnection.setDefaultHostnameVerifier { _, _ -> true }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }
}